src/composables/pendingMultisigTransaction.ts

```diff
@@ -183,28 +183,43 @@ export function usePendingMultisigTransaction() {
       const { data: [latestTransaction] } = await middleware.getTxs({
         direction: 'backward', limit: 1, contract: contractId,
       });
-      latestMultisigAccountTransaction.value = latestTransaction;
+      return latestTransaction;
     } catch (error) {
       handleUnknownError(error);
+      return null;
     }
   }
 
   if (!composableInitialized) {
     composableInitialized = true;
 
-    watch(
-      activeMultisigAccount,
-      (newValue, oldValue) => {
-        if (!isEqual(newValue, oldValue)) {
-          assignPendingMultisigTx();
-
-          if (!activeMultisigAccount.value?.txHash && !latestMultisigAccountTransaction.value) {
-            fetchLatestMultisigAccountTransaction();
+    // Create persistent effect scope to avoid watcher being disposed
+    effectScope(true).run(() => {
```

Comment:
Can you take a look on this solution @CedrikNikita and @martinkaintas ?
This is using this concept: https://github.com/vuejs/rfcs/blob/master/active-rfcs/0041-reactivity-effect-scope.md

The downside of this solution is that we need to initiate watcher dependencies within this custom scope.

Reply:
LGTM, I would also like to see if we can refactor our shared composables to use `effectScope` like [this example](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0041-reactivity-effect-scope.md#example-a-shared-composable)